hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonInputFormat.java

```diff
@@ -504,7 +505,24 @@ public QueryModel createQueryModel(InputSplit inputSplit, TaskAttemptContext tas
     String projectionString = getColumnProjection(configuration);
     String[] projectColumns;
     if (projectionString != null) {
-      projectColumns = projectionString.split(",");
+      if (projectionString.equalsIgnoreCase("*")) {
```

Comment:
instead of passing `*`, I think better to add another function to project all columns. You can add `projectAllColumns()`

Reply:
ok, I will raise new PR for it.